src/importer/gltf.cpp

```diff
@@ -1630,22 +1679,29 @@ static bool gltfImportAssets(LoaderData &loader,
     DynArray<SourceMesh> merged_meshes(total_new_meshes);
     DynArray<math::Vector3> new_positions_arr(total_new_vertices);
     DynArray<math::Vector3> new_normals_arr(total_new_normals);
+    DynArray<math::Vector2> new_uvs_arr(total_new_uvs);
+    DynArray<uint32_t> new_mats_arr(total_new_mats);
```

Comment:
I don't like that all this mat ID stuff has made it all the way down into the importer code. Can this be computed *after* import while setting up the MeshBVH for rendering? Separation of concerns, etc?

Reply:
this is now nuked - no need for it anymore. my bad a lot of this is remnants of some of the experiments we did where we merged the entire scene into one mesh i think. anyway, it's gone now